src/utils.rs

```diff
@@ -0,0 +1,30 @@
+use num_bigint::BigUint;
+use starknet::core::types::FieldElement;
+use anyhow::anyhow;
+
+pub fn felt_to_hex(value: &FieldElement, with_prefix: bool) -> String {
+    match with_prefix {
+        true => format!("0x{}", felt_to_hex(value, false)),
+        false => BigUint::from_bytes_be(&value.to_bytes_be()).to_str_radix(16),
+    }
+}
```

Comment:
There are already other functions that perform this. Could you please refactor the whole code, so that every where it is needed it's the same `felt_to_hex`